server.cpp

```diff
@@ -38,101 +39,31 @@ void connection::stop() {
 void connection::do_read() {
 	auto self(shared_from_this());
   socket_.async_read_some(boost::asio::buffer(buffer_),
+	//TODO: inline specification
       [this, self](boost::system::error_code ec, std::size_t bytes)
       {
-      	request_parser::result_type res;
-        std::tie(res, std::ignore) = request_parser_.parse(request_, buffer_.data(), buffer_.data() + bytes);
-        std::string uri = request_.uri;
-        boost::filesystem::path path_{uri};
-	base_handler* handler = nullptr;
-
-	auto cur_path = (*paths_)[path_.parent_path().string()];
-	if(cur_path == nullptr|| cur_path->options[std::string(FILE_HANDLER_ROOT_TOKEN)] == nullptr)
-		cur_path = (*paths_)[path_.string()];
-	if(cur_path != nullptr)
-	{
-		auto cur_option = cur_path->options[std::string(FILE_HANDLER_ROOT_TOKEN)];
-		std::string doc_root;
-		if(cur_option != nullptr)
-		{
-			request_.uri = "/" + path_.filename().string();
-			doc_root += cur_option->value;
-		}
-		handler = base_handler::make_handler(cur_path->handler_name, doc_root);
-	}
+        request_ = Request::Parse(buffer_.data());
+        std::string uri = request_->uri();
+        boost::filesystem::path path{uri};
+	RequestHandler* handler = nullptr;
+
+	auto pathIt = path.begin();
+	auto cur_prefix = pathIt->string();
+	++pathIt;
+	cur_prefix += pathIt->string();
+	
+	if((*handlers_)[cur_prefix] != nullptr)
+        	(*handlers_)[cur_prefix]->HandleRequest(*request_, &response_);
```

Comment:
This is great and very similar to what we discussed in our common api. But here are we checking to see which handler is appropriate, or are we just trying each of them to see which one will return true. How do we know which handler's HandleRequest function to use given a request.